### PR TITLE
Add Connect-Protocol-Version to CORS configuration

### PR DESCRIPTION
**Pull request by akshayjshah**
Update to match the latest revision to the Connect protocol.

### CODE REVIEW

**Review by joshcarp — APPROVED** [review]
👍

File: main.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ func newCORS() *cors.Cors {
 			"Accept-Post",
 			"Connect-Accept-Encoding",
 			"Connect-Content-Encoding",
+			"Connect-Protocol-Version",
```
Comment (timostamm) [inline main.go:112]: This adds the field name `Connect-Protocol-Version` to the response header `Access-Control-Expose-Headers`. This allows browser clients to access the response header `Connect-Protocol-Version`. But we never send `Connect-Protocol-Version` in responses, right? That means we don't need this change.

**Review by timostamm — COMMENTED** [review]
Our CORS setup is actually already permissive enough that we don't need to change anything here!

The magic happens on this line:

https://github.com/bufbuild/connect-demo/blob/a3b03a93981bca08f73806ffbbb234df8ff6e3bd/main.go#L104

Browser clients attempting to send a `Connect-Protocol-Version` request header across domains will first send a preflight request with `Access-Control-Request-Headers: Connect-Protocol-Version`. 
`rs/cors` will automatically respond with `Access-Control-Allow-Headers : Connect-Protocol-Version`, giving the browser the green light to continue.
Note the subtle but important difference to a preflight response with `Access-Control-Allow-Headers : *` :see_no_evil:

In practice, most deployments will likely be more strict about allowed headers, or not have a facility to automatically allow all incoming headers. `rs/cors` has this feature, but [envoy](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/cors/v3/cors.proto#envoy-v3-api-msg-extensions-filters-http-cors-v3-corspolicy) does not.

**Comment by akshayjshah** [comment]
🤦🏽‍♂️  Clearly, I'm a CORS idiot. Sorry for the spam.

**Pull request closed without merging**